client/src/components/advancedSearch/DateRangeFilter.tsx

```diff
@@ -183,12 +203,16 @@ export const deserialize = ({ queryKey }, query, key) => {
   const filterValue = {}
 
   if (query[startKey]) {
-    toQueryValue[startKey] = query[startKey]
+    const startVal = query[startKey]
+    toQueryValue[startKey] = startVal
     const lastValues = [LAST_DAY, LAST_WEEK, LAST_MONTH]
-    if (lastValues.indexOf(+query[startKey]) !== -1) {
-      filterValue.relative = query[startKey]
+    if (lastValues.indexOf(+startVal) !== -1) {
+      filterValue.relative = startVal
+    } else if (!isNaN(parseInt(startVal))) {
+      filterValue.relative = LAST_X_DAYS
+      filterValue.days = parseInt(startVal)
```

Comment:
This is incorrect, it sets the number of milliseconds as the number of days…
(Try it by saving a search for e.g. "Last 17 days" and showing that saved search.)
Should probably be changed to `filterValue.days = parseInt(startVal) / LAST_DAY`